addons/hw_drivers/tools/helpers.py

```diff
@@ -188,6 +191,22 @@ def save_conf_server(url, token, db_uuid, enterprise_code):
     write_file('odoo-db-uuid.conf', db_uuid or '')
     write_file('odoo-enterprise-code.conf', enterprise_code or '')
 
+def generate_password():
+    """
+    Generate an unique code to secure raspberry pi
+    """
+    alphabet = string.ascii_letters + string.digits
+    password = ''.join(secrets.choice(alphabet) for i in range(12))
+    salt = ''.join(secrets.choice(alphabet) for i in range(12))
+    shadow_password = crypt.crypt(password, '$1$' + salt + '$')
```

Comment:
Nothing particular to say regarding the code, by reading the documentation we can see that the salt generation can off-loaded to the crypt library by using `crypt.mksalt`